File: dcp-solutions/2024-11-09/solution.c
```c
#include <stdbool.h>

bool canPartition(int* nums, int numsSize) {
    int totalSum = 0;
    for (int i = 0; i < numsSize; i++) {
        totalSum += nums[i];
    }
    
    if (totalSum % 2 != 0) return false;
    
    bool dp[totalSum / 2 + 1][numsSize + 1] = {false};
    dp[0][0] = true;

    for (int i = 1; i <= numsSize; i++) {
        for (int j = 1; j <= totalSum / 2; j++) {
            if (j >= nums[i - 1]) {
                dp[j][i] = dp[j][i - 1] || dp[j - nums[i - 1]][i - 1];
            } else {
                dp[j][i] = dp[j][i - 1];
            }
        }
    }

    return dp[totalSum / 2][numsSize];
}
```

File: dcp-solutions/2024-11-09/solution.rb
```ruby
def canPartition(nums)
  total_sum = nums.sum
  return false if total_sum % 2 != 0 || total_sum / 2 < nums.min

  (1..total_sum / 2).each do |target_sum|
    dp = Array.new(total_sum + 1, false)
    dp[0] = true
    nums.each do |num|
      (dp.length - num...dp.length).each do |i|
        next if dp[i] == false
        dp[i + num] = true if i + num <= dp.length && !dp[i + num]
      end
    end

    return true if dp[total_sum / 2]
  end

  false
end
```